modules/sources/enrichers/swaps-usd.ts

```diff
@@ -40,11 +40,12 @@ export async function swapsUsd(swaps: SwapEvent[], chain: Chain): Promise<SwapEv
             const tokenOut = tokenPrices.find((price) => price.tokenAddress === swap.payload.tokenOut.address);
             const feeToken = tokenPrices.find((price) => price.tokenAddress === swap.payload.fee.address);
             const surplusToken = tokenPrices.find((price) => price.tokenAddress === swap.payload.surplus?.address);
+            const feeValueUSD = parseFloat(swap.payload.fee.amount) * (feeToken?.price || 0);
 
             const payload = {
                 fee: {
                     ...swap.payload.fee,
-                    valueUSD: String((feeToken?.price || 0) * parseFloat(swap.payload.fee.amount)),
+                    valueUSD: String(feeValueUSD > 0 ? feeValueUSD : swap.payload.fee.valueUSD),
```

Comment:
I think you set it to `swap.payload.fee.valueUSD` because that is still 0 at this time? Think this is a but bogus, just set to 0?

Reply:
It's a passthrough for a default USD price coming from the subgraph, in case there is no pricing in the DB. It's set earlier and it important to the FX fee, because it's set based on the subgraph data of latestFx rates.

Reply:
Ah gotcha, this is v2. This is just used as fallback so its fine 👍